pkg/generate/code/set_resource.go

```diff
@@ -1229,6 +1238,29 @@ func SetResourceForStruct(
 	for memberIndex, memberName := range sourceShape.MemberNames() {
 		targetMemberShapeRef := targetShape.MemberRefs[memberName]
 		if targetMemberShapeRef == nil {
+			// This can happen when the targetShape is a primitive
+			// and the sourceShape is a struct like when EC2 resource DHCPOptions
+			// needs to set NewDhcpConfiguration.Values (targetShape = string) field from
+			// DhcpConfiguration.Values (sourceShape = AttributeValue). AttributeValue
+			// has a 'Value' member whereas the string shape does not. However,
+			// there is a possibility that the targetShape can be set from
+			// a different member within sourceShape using SetConfig.
+			if currentField, ok := r.Fields[currentPath]; ok {
+				setCfg := currentField.GetSetterConfig(model.OpTypeList)
+				if setCfg != nil && setCfg.From != nil {
+					// If the Field's SetConfig refers to a member in
+					// sourceShape, then update targetMemberShapeRef
+					// to the intended target, targetShape.
+					fp := fieldpath.FromString(*setCfg.From)
+					updatedSourceShapeRef := fp.ShapeRef(sourceShapeRef)
+					if updatedSourceShapeRef != nil && updatedSourceShapeRef.Shape != nil {
+						targetMemberShapeRef = targetShapeRef
+					}
+				}
+			}
+		}
```

Comment:
This block of code does not reference the member field path. `currentPath` will always refer to the containing shape's field path... I think you need to need to build up a `memberFieldPath` containing `currentPath` + '.' + `memberName`?

Reply:
`currentPath` will always refer to current target shape so fetching the `SetConfig` is possible, and the path is being built on line 1295.